pex/dist_metadata.py

```diff
@@ -7,11 +7,27 @@
 import email
 
 from pex.third_party.packaging.specifiers import SpecifierSet
-from pex.third_party.pkg_resources import DistInfoDistribution, Distribution
+from pex.third_party.pkg_resources import DistInfoDistribution, Distribution, Requirement
 from pex.typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from typing import Optional
+    from typing import Dict, Iterator, Optional
+
+
+_PKG_INFO_BY_DIST = {}  # type: Dict[Distribution, Optional[email.message.Message]]
+
+
+def _parse_pkg_info(dist):
+    # type: (Distribution) -> Optional[email.message.Message]
+    global _PKG_INFO_BY_DIST
```

Comment:
Because you don't assign to `_PKG_INFO_BY_DIST` (only internally mutate it), the global keyword shouldn't be necessary here?

I guess that in this case this is an alternative to the `memoized` decorator?

Reply:
The `memoized` is a Pants thing, not a Pex thing, but you're correct - will fix.